sqlglot/parser.py

```diff
@@ -2242,7 +2242,12 @@ def _parse_lateral(self) -> t.Optional[exp.Lateral]:
             return None
 
         if not this:
-            this = self._parse_function() or self._parse_id_var(any_token=False)
+            this = (
+                self._parse_unnest()
```

Comment:
Without this we used to parse the `UNNEST` in `... JOIN LATERAL UNNEST(...)` into `exp.Anonymous`.